desktop-widgets/tab-widgets/TabDiveInformation.cpp

```diff
@@ -188,6 +187,7 @@ void TabDiveInformation::updateData()
 		ui->waterTypeText->clear();
 	}
 	updateMode(current_dive);
+	updateSalinity();
```

Comment:
Now this code is run twice. Once in the copied version, and once in the function called.

Reply:
My misunderstanding. Fixed now.

Reply:
Move the updateSalinity back where it were before, then, if there is no reason for moving it.

Reply:
Also, the whole waterTypeText block should go into the updateSalinity method, so all calls to updateSalinity also refreshes the waterTypeText. Otherwise they will diverge on divesChanged.

Reply:
This is a good suggestion that I implemented. Thank you, @glance

Reply:
Not seeing that code, did you forget to push?

Reply:
The previous push did not go as expected. Please look now? solved?